worker/uniter/remotestate/watcher.go

```diff
@@ -389,25 +384,15 @@ func (w *RemoteStateWatcher) loop(unitTag names.UnitTag) (err error) {
 		instanceDataChannel     watcher.NotifyChannel
 	)
 
-	// CAAS models don't use an application watcher
-	// which fires an initial event.
-	if w.modelType == model.CAAS && !w.sidecar {
-		seenApplicationChange = true
+	applicationw, err := w.application.Watch()
+	if err != nil {
+		return errors.Trace(err)
 	}
-
-	if w.modelType == model.IAAS || w.sidecar {
-		// For IAAS model and sidecar CAAS model, we need to watch state for
-		// application charm changes instead of being informed by the operator.
-		applicationw, err := w.application.Watch()
-		if err != nil {
-			return errors.Trace(err)
-		}
-		if err := w.catacomb.Add(applicationw); err != nil {
-			return errors.Trace(err)
-		}
-		w.applicationChannel = applicationw.Changes()
-		requiredEvents++
+	if err := w.catacomb.Add(applicationw); err != nil {
+		return errors.Trace(err)
 	}
+	w.applicationChannel = applicationw.Changes()
```

Comment:
This doesn't need to be an attribute on the watcher struct anymore. It can just be a function local variable like the other watchers are done.